app/ts/components/Transfer.tsx

```diff
@@ -1,177 +1,194 @@
-									{selectedNft.value ? <p className='truncate w-full'>{selectedNft.value.name}</p> : <p className='w-18 h-4 rounded bg-white/20 animate-pulse'></p>}
-								</div>
-								<div>
-									<span className='text-sm text-white/50'>Token ID</span>
-									{selectedNft.value ? <p className='truncate w-full'>{selectedNft.value.id}</p> : <p className='w-18 h-4 rounded bg-white/20 animate-pulse'></p>}
-								</div>
-								<div className='w-full'>
-									<span className='text-sm text-white/50'>Token Metadata</span>
-									{selectedNft.value ? <a className='truncate w-full block hover:underline' target='_blank' href={selectedNft.value.tokenURI}>{selectedNft.value.tokenURI}</a> : <p className='w-18 h-4 rounded bg-white/20 animate-pulse'></p>}
-								</div>
-							</div>
-							<div className='flex flex-col gap-4 flex-1 w-full max-w-xl'>
-								<div>
-									<span className='text-sm text-white/50'>Contract Address</span>
-									{selectedNft.value ? <span className='truncate w-full flex items-center gap-2'><Blockie seed={selectedNft.value.address.toLowerCase()} size={4} />{selectedNft.value.address}</span> : <p className='w-18 h-4 rounded bg-white/20 animate-pulse'></p>}
-								</div>
-								<div>
-									<span className='text-sm text-white/50'>Owner</span>
-									{selectedNft.value ? <span className='truncate w-full flex items-center gap-2'><Blockie seed={selectedNft.value.owner.toLowerCase()} size={4} />{selectedNft.value.owner}</span> : <p className='w-18 h-4 rounded bg-white/20 animate-pulse'></p>}
-								</div>
-							</div>
-						</>
-						: null}
-					{openseaParseError.value ? <p>{openseaParseError.value}</p> : null}
-					{fetchingStates.value === 'notfound' || fetchingStates.value === 'contract' ? <p>No NFT found at address</p> : null}
-					{fetchingStates.value === 'badid' ? <p>Found NFT collection, but token ID does not exist</p> : null}
-					{fetchingStates.value === 'noprovider' ? <p>Connect wallet to load token details.</p> : null}
-					{fetchingStates.value === 'EOA' ? <p>Address provided is an EOA</p> : null}
-					{fetchingStates.value === 'ERC20' ? <p>Address provided is an ERC20 contract</p> : null}
-					{fetchingStates.value === 'ERC1155' ? <p>Address provided is an ERC1155 contract</p> : null}
-				</div>)}
-			<BlockieTextInput onInput={validateRecipientInput} label='Recipient Address' warn={showWarn.value.recipient} placeholder='0x133...789' />
+					{fetchingStates.value === 'fetching' ? <ItemDetails item={selectedNft} /> : null}
+				</div>
+			) : null}
+			<BlockieTextInput value={recipientInput} label='Recipient Address' warn={showWarn.value.recipient} placeholder='0x133...789' />
+			{warning.value ? (
+				<div class='flex items-center items-center border border-orange-400/50 bg-orange-400/10 px-4 py-2 gap-4'>
+					<svg width='1em' height='1em' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg' className='text-4xl'>
```

Comment:
the svg icon could be moved to other file and then referenced here

Reply:
For single use, I like inline personally.  I wouldn't be opposed to extracting to a separate file though.